llvm/lib/Target/RISCV/RISCVInstrInfo.cpp

```diff
@@ -5190,3 +5190,33 @@ bool RISCVInstrInfo::isHighLatencyDef(int Opc) const {
     return true;
   }
 }
+
+bool RISCVInstrInfo::isVRegCopy(const MachineInstr *MI, unsigned LMUL) const {
+  if (MI->getOpcode() != TargetOpcode::COPY)
+    return false;
+  const MachineRegisterInfo &MRI = MI->getMF()->getRegInfo();
+  const TargetRegisterInfo *TRI = MRI.getTargetRegisterInfo();
+
+  Register DstReg = MI->getOperand(0).getReg();
+  const TargetRegisterClass *RC = DstReg.isVirtual()
+                                      ? MRI.getRegClass(DstReg)
+                                      : TRI->getMinimalPhysRegClass(DstReg);
+
+  switch (LMUL) {
```

Comment:
Can we use `RISCVRegisterInfo::isVRRegClass`, `RISCVRI::getLMul`, and `RISCVVType::decodeVLMUL`?